### PR TITLE
MGMT-1005 Allow downloading files while the cluster is in error.

### PR DESCRIPTION
**Pull request by eranco74**

### CODE REVIEW

File: internal/bminventory/inventory.go [hunk 1]
```diff
@@ -1147,8 +1148,10 @@ func (b *bareMetalInventory) DownloadClusterFiles(ctx context.Context, params in
 		}
 	}
 	clusterStatus := swag.StringValue(cluster.Status)
-	if clusterStatus != ClusterStatusInstalling && clusterStatus != ClusterStatusInstalled {
-		msg := fmt.Sprintf("Cluster %s is in %s state, files can be downloaded only in installing or installed state", params.ClusterID, clusterStatus)
+	allowedStatuses := []string{ClusterStatusInstalling, ClusterStatusInstalled, ClusterStatusError}
+	if !funk.ContainsString(allowedStatuses, clusterStatus) {
```
Comment (filanov) [inline internal/bminventory/inventory.go:1152]: why is it not part of cluster api? the states are internal to cluster inventory.go is not related to cluster states.

Comment (eranco74) [reply internal/bminventory/inventory.go:1152]: Because I don't want to start refactoring while we are in code freeze.

**Comment by eranco74** [comment]
Subsystem:
```
Ran 22 of 22 Specs in 85.635 seconds
SUCCESS! -- 22 Passed | 0 Failed | 0 Pending | 0 Skipped
```

File: internal/bminventory/inventory.go (hunk 1 above)
Comment (eranco74) [reply internal/bminventory/inventory.go:1152]: https://github.com/filanov/bm-inventory/pull/212

File: internal/bminventory/inventory.go [hunk 2]
```diff
@@ -1147,8 +1148,10 @@ func (b *bareMetalInventory) DownloadClusterFiles(ctx context.Context, params in
 		}
 	}
 	clusterStatus := swag.StringValue(cluster.Status)
-	if clusterStatus != ClusterStatusInstalling && clusterStatus != ClusterStatusInstalled {
-		msg := fmt.Sprintf("Cluster %s is in %s state, files can be downloaded only in installing or installed state", params.ClusterID, clusterStatus)
+	allowedStatuses := []string{ClusterStatusInstalling, ClusterStatusInstalled, ClusterStatusError}
```
Comment (ronniel1) [inline internal/bminventory/inventory.go:1151]: nice

File: internal/bminventory/inventory.go (hunk 1 above)
Comment (filanov) [reply internal/bminventory/inventory.go:1152]: you told me that same last time :)

**Pull request merged**